### PR TITLE
[Test Only] Add fp32 sort index test

### LINKED ISSUE #26057: sort operation produces incorrect values and indices
### Describe the bug

When performing sort operations on the same input tensor, TTNN and PyTorch produce completely different results:
Values Mismatch

PyTorch values: Range from -inf to 22.1406
TTNN values: Range from -131008.0 to 131008.0

Indices Mismatch

PyTorch indices: Valid tensor indices (e.g., 101298, 101289, 101290, ..., 1416, 2014, 17097)
TTNN indices: Invalid/corrupted indices with negative values (e.g., 5, 4, 6, ..., -31178, -31180, -31179)

The test case loads input data from a NumPy file (attched scores.npy) and compares TTNN experimental sort against PyTorch's native sort:

```
# Load input tensor from file
torch_a = torch.from_numpy(np.load('scores.npy'))

# Convert to TTNN with TILE_LAYOUT 
a = ttnn.from_torch(torch_a, layout=ttnn.TILE_LAYOUT, device=device)

# Compare outputs
ttnn_output = ttnn.experimental.sort(a, descending=False)
torch_values, torch_indices = torch.sort(torch_a, descending=False)
```

[ttnn_sort.txt](https://github.com/user-attachments/files/21529168/ttnn_sort.txt)

[scores.txt](https://github.com/user-attachments/files/21529179/scores.txt)

### Steps to reproduce the issue

_No response_

### Expected behavior

_No response_

### Please complete the following environment information

_No response_

### PR DESCRIPTION
**Pull request by iwroszTT**
### PR Category
Test Only

### Summary
Relates to #26057. Adds regression coverage for its repro (fp32 `[1, 151936]` logits, mostly `-inf`), whose reported defects — fp32 value saturation and int16 index overflow — are fixed on main (#41007, #33983) but had no test pinning them: the closest existing case (`test_sort_long_tensor`) covers the shape in bfloat16 and asserts values only.

The new test asserts, per direction:
- sorted values match `torch.sort` exactly (pins the fp32 saturation defect),
- indices are uint32 (pins the int16 overflow defect),
- indices are in range and form a valid permutation,
- gathering the input by the returned indices reproduces the sorted values.

Exact torch index parity is deliberately not asserted: the input is tie-heavy and ties sort unstably until #25797/#33492 issues are resolved.

`descending=True` is an xfail for #53326 (found while writing this test): the `-inf` tile padding ties with real `-inf` entries and out-of-range indices go into the output.

### Notes for reviewers
- Runtime is ~2s for both parametrizations on Blackhole p100a (warm JIT cache): one shape, two directions.

### CI Status
_Auto-generated on every push. Badges update live. Click a badge to filter runs by this branch._

- [](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:iwroszTT/26057-sort-fp32-wide-index-regression)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/runtime-sanity-tests.yaml?query=branch:iwroszTT/26057-sort-fp32-wide-index-regression)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:iwroszTT/26057-sort-fp32-wide-index-regression)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/all-model-tests.yaml?query=branch:iwroszTT/26057-sort-fp32-wide-index-regression)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:iwroszTT/26057-sort-fp32-wide-index-regression)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:iwroszTT/26057-sort-fp32-wide-index-regression)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:iwroszTT/26057-sort-fp32-wide-index-regression)

### CODE REVIEW

File: tests/ttnn/unit_tests/operations/data_movement/test_sort.py [hunk 1]
```diff
@@ -187,6 +187,42 @@ def test_sort_long_tensor(shape, dim, descending, device):
...
+    ttnn_sort_values, ttnn_sort_indices = ttnn.sort(ttnn_input, dim=-1, descending=descending)
+
+    assert ttnn_sort_indices.dtype == ttnn.uint32
+
+    values = ttnn.to_torch(ttnn_sort_values)
+    indices = ttnn.to_torch(ttnn_sort_indices).reshape(-1).to(torch.int64)
+
+    assert_equal(torch_sort_values, values)
+    assert indices.min() >= 0 and indices.max() < n
+    assert indices.unique().numel() == n
```
[the remark this answers is not in the document]
Comment (iwroszTT) [reply tests/ttnn/unit_tests/operations/data_movement/test_sort.py]: Improved the error message

File: tests/ttnn/unit_tests/operations/data_movement/test_sort.py [hunk 2]
```diff
@@ -187,6 +187,42 @@ def test_sort_long_tensor(shape, dim, descending, device):
         assert_equal(torch_sort_values, ttnn.to_torch(ttnn_sort_values))
 
 
+@pytest.mark.parametrize(
+    "descending",
+    [
+        False,
+        # Descending order pads the row with -inf, which ties with real -inf
+        # entries and can emit padding indices (>= n) into the output.
+        pytest.param(True, marks=pytest.mark.xfail(strict=True, reason="#53326: padding indices leak")),
```
[the remark this answers is not in the document]
Comment (iwroszTT) [reply tests/ttnn/unit_tests/operations/data_movement/test_sort.py]: Restructured into two test cases

**Review by itarabanTT — APPROVED** [review]
